homeassistant/components/ai_speaker/__init__.py

```diff
@@ -0,0 +1,48 @@
+"""Support for AI-Speaker."""
+import logging
+
+from homeassistant.helpers import aiohttp_client
+
+from .const import AIS_WS_COMMAND_URL, DOMAIN
+
+PLATFORMS = ["sensor", "media_player"]
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass, config):
+    """Set up the initial domain configuration."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up the integration based on a configuration entry."""
+
+    _LOGGER.debug("async_setup_entry %s", config_entry)
+    web_session = aiohttp_client.async_get_clientsession(hass)
+
+    async def async_command(service):
+        """Publish command to AI-Speaker WS."""
+        requests_json = {service.data["key"]: service.data["val"]}
+        ais_ws_url = AIS_WS_COMMAND_URL.format(ais_url=service.data["ais_url"])
+        try:
+            await web_session.post(ais_ws_url, json=requests_json, timeout=3)
```

Comment:
Please extract the device specific interface code and data parsing to a standalone library published on PyPI.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices

Reply:
@MartinHjelmare, done in #32b2f2d thanks - this move simplified the code a lot :)